### DISCUSSION TITLE
Text extraction excluding Image and Tables

### DISCUSSION THREAD

**Discussion by ytiam** [jsvine/pdfplumber#614] [Q&A]
Hi guys,
Is there any trick or method available, if anyone wants to just extract the textual information available just inside the paragraph lines, excluding the texts available inside Image and Table? If anyone knows any logic, please help with that as well.

**Comment by jsvine** [comment]
Hi @ytiam, and thanks for your interest in this library. I'm not quite sure what you mean about text inside an "Image", but here's one way you could exclude text within a table:

```python
from pdfplumber.utils import intersects_bbox

def get_nontable_text(page):
  tables = page.find_tables()

  def outside_tables(obj):
      return not any(intersects_bbox([obj], t.bbox) for t in tables)

  return page.filter(outside_tables).extract_text()
```

**Reply by ytiam** [reply]
Thanks for your reply and solution @jsvine, I will try this and update you. As I was in hurry for project deadlines and the implementation, I got an alternative solution from stackoverflow and it is typically based on similar kind of concepts related to bounding box. 

Link: https://stackoverflow.com/questions/69407850/pdfplumber-extract-text-function-also-extracts-text-from-the-table-only-want-to

I developed my custom code based on this and it is working fine for both tables and images.

**Reply by jsvine** [reply]
Great, and thanks for the update!
